rtCommon/bidsIncremental.py

```diff
@@ -0,0 +1,451 @@
+            return False
+
+        # Compare image metadata
+        if self._imgMetadata != other._imgMetadata:
+            reportDifference("Image metadata",
+                             self._imgMetadata,
+                             other._imgMetadata,
+                             np.array_equal)
+            return False
+
+        # Compare dataset metadata
+        if self.datasetMetadata != other.datasetMetadata:
+            reportDifference("Dataset metadata",
+                             self.datasetMetadata,
+                             other.datasetMetadata)
+            return False
+
+        if not np.array_equal(self.imageData(), other.imageData()):
+            differences = self.imageData() != other.imageData()
+            logger.debug("Image data didn't match")
+            logger.debug("Difference count: %d (%f%%)",
+                         np.sum(differences),
+                         np.sum(differences) / np.size(differences) * 100.0)
+            return False
+
+        return True
+
+    @staticmethod
+    def createImageMetadataDict(subject: str, task: str, suffix: str,
+                                repetitionTime: int, echoTime: int):
+        """
+        Creates an image metadata dictionary for a BIDS-I with all of the
+        basic required fields using the correct key names.
+
+        Args:
+            subject: Subject ID (e.g., '01')
+            task: Task ID (e.g., 'story')
+            suffix: Data type (e.g., 'bold')
+            repetitionTime: TR time, in seconds, used for the imaging run
+            echoTime: Echo time, in seconds, used for the imaging run
+
+        Returns:
+            Dictionary with the provided information ready for use in a BIDS-I
+
+        """
+        return {"subject": subject, "task": task, "suffix": suffix,
+                "RepetitionTime": repetitionTime, "EchoTime": echoTime}
+
+    @classmethod
+    def missingImageMetadata(cls, imageMeta: dict) -> list:
+        return [f for f in cls.REQUIRED_IMAGE_METADATA if f not in imageMeta]
+
+    @classmethod
+    def isCompleteImageMetadata(cls, imageMeta: dict) -> bool:
+        """
+        Verifies that all required metadata fields for BIDS-I construction are
+        present in the dictionary.
+
+        Args:
+            imageMeta: The dictionary with the metadata fields
+
+        Returns:
+            True if all required fields are present in the dictionary, False
+            otherwise.
+
+        """
+        return len(cls.missingImageMetadata(imageMeta)) == 0
+
+    def _exceptIfNotBids(self, entityName: str):
+        """ Raise an exception if the argument is not a valid BIDS entity """
+        if self.ENTITIES.get(entityName) is None:
+            raise ValueError(f"{entityName} is not a valid BIDS entity name")
+
+    # TODO(spolcyn): Add specific getters for commonly used things, like getRun,
+    # getSubject, getTask
+    def getMetadataField(self, field: str, strict: bool = False) -> str:
```

Comment:
Is there a list of fields in the code or doc-strings?

Reply:
Field could be anything; added some examples to the docstring.